chore(meta): Add info to writing guidelines about BCD definitions of experimental and obsolete (#22575)
mdn/content: files/en-us/mdn/writing_guidelines/experimental_deprecated_obsolete/index.md

These terms are commonly associated with technologies and specifications and are used on MDN Web Docs to label the status of a technology. For the definition of these terms, MDN Web Docs align with the [Browser Compatibility Data (BCD)](https://github.com/mdn/browser-compat-data/blob/main/schemas/compat-data-schema.md#status-information) repository.
These terms are described below in the context of their use on MDN Web Docs.

## Experimental

When a technology is described as experimental on MDN Web Docs, it means that the technology is nascent and immature and is currently _in the process_ of being added to the web platform (or being considered for addition).
Marking a technology as experimental indicates that readers should think carefully before using that technology in any kind of production project (i.e., a project that is not just a demo or experiment). Readers are [encouraged to try out experimental features](https://github.com/mdn/browser-compat-data/blob/main/schemas/compat-data-schema.md#status-information) and provide feedback to browser vendors and standards authors.

For a technology marked **experimental**, one or more of the following conditions apply:

> - It is implemented and enabled by default in _one and only one_ modern major browser rendering engine.

Reviewer: Just IMO
```suggestion
- It is implemented and enabled by default in **only one** modern major browser rendering engine.
```

> - It is only supported by making configuration changes such through preferences or flags, regardless of the number of supported rendering engines.

Reviewer: I'd be tempted to pull this out as a note. It is accurate but the point is "preferenced or betafeatures do not count towards something being experimental or not" - and that is hard to read from this.

Something like:

> **Note:** A feature that is only released on one rendering engine is still considered experimental even if it is available on preview builds of other rendering engines (or by setting a preference or flag).

Author: The point is moreso that something is experimental if it's behind a preference even if that's true for multiple browsers. So we cover `experimental = true` for these:

* supported only behind a pref in one or more browsers
* supported by default in only one browser
* supported by default in one browser, behind a pref in others

Your note still applies, though, so I would add it.

- Its defining specification is likely to change significantly in backwards-incompatible ways (i.e., it may break existing code that relies on the feature).

The **experimental** status of a technology expires if one or more of the following conditions is met:

- It is supported by default in _two or more_ major browser rendering engines.

> - It is supported by default by a browser rendering engine for two or more years and undergoes no major changes.

Reviewer: Does this mean "any" engine? I.e. in chrome for two years but not any other engine it becomes standard automatically?

You might want to clarify "a single" browser rendering engine.

This was news to me :-)

Author: > Does this mean "any" engine? I.e. in chrome for two years but not any other engine it becomes standard automatically?

:-) Yes, this is the BCD way that I wanted to capture in this PR -> https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental

The term **deprecated** on MDN Web Docs is used to mark an API or technology that is no longer recommended. A deprecated API or technology might be removed in the future or might only be kept for compatibility purposes and may still work. We recommend to avoid using the functionality marked as deprecated.

For more information on the definition of **deprecated**, see the [deprecated flag](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated) BCD documentation.

## Obsolete

On MDN Web Docs, the term **obsolete** was historically used to indicate an API or technology that is not only no longer recommended but is also no longer implemented in browsers.

The distinction between **obselete** and **deprecated** is not very helpful.
We do not use the term **obsolete** on MDN Web Docs anymore.

>  > **Note:** If you come across any instance of **obselete**, they should be removed or replaced with the term **deprecated**.

Reviewer: FYI, decided to tidy this up. You might want to modify further.

Author: Thanks a lot 👍🏻

The term **deprecated** on MDN Web Docs is used to mark an API or technology that is no longer recommended. A deprecated API or technology might be removed in the future or might only be kept for compatibility purposes and may still work. We recommend to avoid using the functionality marked as deprecated.

For more information on the definition of **deprecated**, see the [deprecated flag](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated) BCD documentation.

## Obsolete

> On MDN Web Docs, the term **obsolete** was used to mark an API or technology that is not only no longer recommended but is also no longer implemented in browsers. The term was, however, confusing. It was similar to **deprecated** but its distinction from **deprecated** was not very helpful.

Reviewer: ```suggestion
On MDN Web Docs, the term **obsolete** was historically used to indicate an API or technology that is not only no longer recommended but is also no longer implemented in browsers. The distinction from **deprecated** is not very helpful.
```

The term **deprecated** on MDN Web Docs is used to mark an API or technology that is no longer recommended. A deprecated API or technology might be removed in the future or might only be kept for compatibility purposes and may still work. We recommend to avoid using the functionality marked as deprecated.

For more information on the definition of **deprecated**, see the [deprecated flag](https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-deprecated) BCD documentation.

## Obsolete

> On MDN Web Docs, the term **obsolete** was historically used to indicate an API or technology that is not only no longer recommended but is also no longer implemented in browsers. The distinction from **deprecated** is not very helpful.
> 
> > **Note:** We do not use the term **obsolete** on MDN Web Docs anymore. If you come across any instances, they should be removed or replaced with the term **deprecated**.

Reviewer: ```suggestion
On MDN Web Docs, the term **obsolete** was historically used to indicate an API or technology that is not only no longer recommended but is also no longer implemented in browsers.

The distinction between **obselete** and **deprecated** is not very helpful.
We do not use the term **obsolete** on MDN Web Docs anymore.

 > **Note:** If you come across any instance of **obselete**, they should be removed or replaced with the term **deprecated**.
```
